subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1484,6 +1484,40 @@ static int cmd_wifi_ps_wakeup_mode(const struct shell *sh, size_t argc, char *ar
 	return 0;
 }
 
+static int cmd_wifi_set_rts(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct net_if *iface = net_if_get_first_wifi();
+	long rts;
+	int rts_threshold = -1;
+
+	context.sh = sh;
+
+	if (strcmp(argv[1], "off") == 0) {
+		rts = -1;
+	} else {
+		if (!parse_number(sh, &rts, argv[1],
+				  WIFI_RTS_THRESHOLD_MIN,
+				  WIFI_RTS_THRESHOLD_MAX)) {
+			return -EINVAL;
+		}
+	}
+
+	rts_threshold = rts;
```

Comment:
I think we could simplify this a bit. For example something like this could be done.
```
int rts = -1; /* default value if user supplies "off" argument */

context.sh = sh;

if (strcmp(argv[1], "off") != 0) {
	long rts_val;

	if (!parse_number(sh, &rts_val, argv[1]),
			  WIFI_RTS_THRESHOLD_MIN,
			  WIFI_RTS_THRESHOLD_MAX)) {
		return -EINVAL;
	}

	if (rts_val < 0) {
		return -EINVAL;
	}

	rts = (int)rts_val;
}
```

Reply:
Updated,  
Below code check is not needed as we have a check for min & max just above that
"if (rts_val < 0) {
		return -EINVAL;
	}
"

Reply:
Indeed, good point.